### PR TITLE
Add installed_extensions prometheus metric

### PR DESCRIPTION
**Pull request by lubennikovaav**
and add /metrics endpoint to compute_ctl to expose such metrics

metric format example for extension pg_rag
with versions 1.2.3 and 1.4.2
installed in 3 and 1 databases respectively:

neon_extensions_installed{extension="pg_rag", version="1.2.3"} = 3 neon_extensions_installed{extension="pg_rag", version="1.4.2"} = 1

------
infra part: https://github.com/neondatabase/flux-fleet/pull/251

revival of the https://github.com/neondatabase/neon/pull/9375
based on this [slack discussion](https://neondb.slack.com/archives/C039YKBRZB4/p1730374730611909?thread_ts=1730286473.132959&cid=C039YKBRZB4):

### CODE REVIEW

**Review by lassizci — APPROVED** [review]
Looks good to me!

File: test_runner/regress/test_installed_extensions.py [hunk 1]
```diff
@@ -85,3 +87,52 @@ def test_installed_extensions(neon_simple_env: NeonEnv):
...
+        try:
+            res = client.metrics()
+            timeout = -1
+            if len(parse_metrics(res).query_all("installed_extensions")) < 4:
+                # Assume that not all metrics that are collected yet
+                time.sleep(1)
+                timeout -= 1
+                continue
+        except Exception as e:
+            info("failed to get metrics, assume they are not collected yet: %s", e)
```
Comment (tristan957) [inline test_runner/regress/test_installed_extensions.py:119]: There is `log.exception` method.

File: test_runner/regress/test_installed_extensions.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import time
 from logging import info
 
+from fixtures.metrics import parse_metrics
 from fixtures.neon_fixtures import NeonEnv
```
Comment (tristan957) [inline test_runner/regress/test_installed_extensions.py:5]:
```suggestion
import time
from logging import info
from typing import TYPE_CHECKING

from fixtures.metrics import parse_metrics

if TYPE_CHECKING:
    from fixtures.neon_fixtures import NeonEnv
```

File: test_runner/regress/test_installed_extensions.py [hunk 3]
```diff
@@ -85,3 +87,52 @@ def test_installed_extensions(neon_simple_env: NeonEnv):
...
+    assert len(neon_m) == 1
+    for sample in neon_m:
+        assert sample.value == 2
+    neon_m = m.query_all("installed_extensions", {"extension_name": "neon", "version": "1.3"})
+    assert len(neon_m) == 1
+    for sample in neon_m:
+        assert sample.value == 1
+
+    endpoint.stop()
+    endpoint.start()
```
Comment (tristan957) [inline test_runner/regress/test_installed_extensions.py:106]: Maybe we can add a restart() method in the future.

Comment (lubennikovaav) [reply test_runner/regress/test_installed_extensions.py:110]: maybe.
but I think current code is readable enough, so this is not my top priority.

File: test_runner/fixtures/endpoint/http.py [hunk 4]
```diff
@@ -46,3 +46,8 @@ def set_role_grants(self, database: str, role: str, schema: str, privileges: lis
         )
         res.raise_for_status()
         return res.json()
+
+    def metrics(self):
```
Comment (tristan957) [inline test_runner/fixtures/endpoint/http.py:50]:
```suggestion
    def metrics(self) -> str:
```

File: test_runner/regress/test_installed_extensions.py [hunk 5]
```diff
@@ -85,3 +91,52 @@ def test_installed_extensions(neon_simple_env: NeonEnv):
...
+        try:
+            res = client.metrics()
+            timeout = -1
+            if len(parse_metrics(res).query_all("installed_extensions")) < 4:
+                # Assume that not all metrics that are collected yet
+                time.sleep(1)
+                timeout -= 1
+                continue
+        except Exception as e:
+            log.exception("failed to get metrics, assume they are not collected yet: %s", e)
```
Comment (tristan957) [inline test_runner/regress/test_installed_extensions.py:123]: When you use `log.exception` you don't need to pass e as as argument in the format string. It picks up the exception from the runtime environment

Comment (tristan957) [reply test_runner/regress/test_installed_extensions.py:123]: https://docs.python.org/3/library/logging.html#logging.Logger.exception

Comment (lubennikovaav) [reply test_runner/regress/test_installed_extensions.py]: thank you, fixed

File: test_runner/regress/test_installed_extensions.py [hunk 6]
```diff
@@ -85,3 +91,52 @@ def test_installed_extensions(neon_simple_env: NeonEnv):
...
+    while timeout > 0:
+        try:
+            res = client.metrics()
+            timeout = -1
+            if len(parse_metrics(res).query_all("installed_extensions")) < 4:
+                # Assume that not all metrics that are collected yet
+                time.sleep(1)
+                timeout -= 1
+                continue
+        except Exception as e:
```
Comment (lubennikovaav) [inline test_runner/regress/test_installed_extensions.py:122]:
```suggestion
        except Exception:
```

File: test_runner/regress/test_installed_extensions.py [hunk 7]
```diff
@@ -1,6 +1,12 @@
+import time
```
Comment (lubennikovaav) [inline test_runner/regress/test_installed_extensions.py:1]:
```suggestion
from __future__ import annotations

import time
```

**Review by tristan957 — APPROVED** [review]

**Pull request merged**